lib/_http_client.js

```diff
@@ -197,7 +201,27 @@ function ClientRequest(input, options, cb) {
   }
   this.insecureHTTPParser = insecureHTTPParser;
 
-  this.path = options.path || '/';
+  path = options.path;
+  if (path) {
+    if (method === 'CONNECT') {
+      if (path.charAt(0) === '/') {
+        path = path.slice(1) || '/';
+        if (!isValidCONNECTPath(path)) {
+          throw new ERR_INVALID_ARG_VALUE('options.path',
+                                          path,
+                                          'must be a valid host:port combo');
+        }
+        this.path = path;
+      } else {
+        this.path = path;
+      }
+    } else {
+      this.path = path;
+    }
+  } else {
+    this.path = '/';
+  }
```

Comment:
Can you assign to `this.path` just once outside the if block? Every `this.path = ...` forces V8 to emit a type check that `this` is an object of the expected shape.

V8 can hoist and coalesce such checks but... it's complicated. Just assign once, please. :-)